lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt

```diff
@@ -70,6 +76,36 @@ abstract class HeanCms(
         .add("Origin", baseUrl)
         .add("Referer", "$baseUrl/")
 
+    private val authHeaders: Headers by lazy {
+        val builder = headersBuilder()
+        if (preferences.loginPref && enableLogin) {
+            val token = getToken()
+            if (token != null) {
+                builder.add("Authorization", "Bearer $token")
+            }
+        }
+        builder.build()
+    }
+
+    private fun getToken(): String? {
+        val body = HeanCmsLoginPayloadDto(preferences.user, preferences.password)
+            .let { json.encodeToString(listOf(it)) }
+            .toRequestBody()
+
+        val loginHeaders = headersBuilder()
+            .add("Next-Action", "cecce9cf69dc6eb35c5b01f633f6b1032d38c61b")
+            .build()
+
+        val response = client.newCall(POST("$baseUrl/login", loginHeaders, body)).execute()
```

Comment:
they appear to have `/login` endpoint in the api, which returns the token and expiration date; just post the `HeanCmsLoginPayloadDto` without list.